examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,851 @@
+      "\n",
+      "struct dataobj\n",
+      "{\n",
+      "  void *restrict data;\n",
+      "  int * size;\n",
+      "  int * npsize;\n",
+      "  int * dsize;\n",
+      "  int * hsize;\n",
+      "  int * hofs;\n",
+      "  int * oofs;\n",
+      "} ;\n",
+      "\n",
+      "struct profiler\n",
+      "{\n",
+      "  double section0;\n",
+      "} ;\n",
+      "\n",
+      "\n",
+      "int Kernel(struct dataobj *restrict f_vec, struct profiler * timers, const int x_M, const int x_m, const int y_M, const int y_m, const int nthreads)\n",
+      "{\n",
+      "  float (*restrict f)[f_vec->size[1]] __attribute__ ((aligned (64))) = (float (*)[f_vec->size[1]]) f_vec->data;\n",
+      "  /* Flush denormal numbers to zero in hardware */\n",
+      "  _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);\n",
+      "  _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);\n",
+      "  struct timeval start_section0, end_section0;\n",
+      "  gettimeofday(&start_section0, NULL);\n",
+      "  /* Begin section0 */\n",
+      "  #pragma omp parallel num_threads(nthreads)\n",
+      "  {\n",
+      "    #pragma omp for collapse(1) schedule(static,1)\n",
+      "    for (int x = x_m; x <= x_M; x += 1)\n",
+      "    {\n",
+      "      #pragma omp simd aligned(f:32)\n",
+      "      for (int y = y_m; y <= y_M; y += 1)\n",
+      "      {\n",
+      "        f[x + 1][y + 1] = f[x + 1][y + 1] + 1;\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  /* End section0 */\n",
+      "  gettimeofday(&end_section0, NULL);\n",
+      "  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(op.ccode) # Print the generated code"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Devito API for relationals\n",
+    "\n",
+    "In order to construct a `ConditionalDimension`, one needs to get familiar with relationals. The Devito API for relationals currently supports the following classes of relations, which inherit from the [SymPy ones](https://docs.sympy.org/latest/modules/core.html#module-sympy.core.relational):\n",
+    "\n",
+    "- Le (less-than or equal)\n",
+    "- Lt (less-than)\n",
+    "- Ge (greater-than or equal)\n",
+    "- Gt (greater-than)\n",
+    "- Ne (not equal)\n",
+    "\n",
+    "Relationals are used to define a condition and are passed as an argument to ConditionalDimension at construction time.\n",
+    "\n",
+    "\n",
+    "# Devito API for ConditionalDimension\n",
+    "\n",
+    "A `ConditionalDimension` defines a non-convex iteration sub-space derived from a ``parent`` Dimension, implemented by the compiler generating conditional \"if-then\" code within the parent Dimension's iteration space.\n",
+    "\n",
+    "A `ConditionalDimension` is used in two typical cases. \n",
+    "\n",
+    "- Use case A:\n",
```

Comment:
Would remove `ci = ConditionalDimension(name, parent, condition)` (can show and explain later during examples) and clean up the language a bit, i.e. `Use case B: To subsample a `Function`.`

Also, current formatting seems a bit off (one of the `ci` expressions not in a bullet).

Reply:
@rhodrin are you fine with this?